content/swift/concepts/functions/functions.md

```diff
@@ -0,0 +1,171 @@
+---
+Title: 'Functions' 
+Description: 'A functions is a set of code organized together to preform a specific task.'
+Subjects: 
+  - 'iOS'
+  - 'Code Foundations'
+Tags:
+  - 'iOS'
+  - 'Algorithms'
+  - 'Functions'
+CatalogContent:
+  - 'learn-swift'
+  - 'paths/build-ios-apps-with-swiftui'
+---
+
+**Functions** are self-contained chunks of code that perform a specific task. A function is given a name that identifies what it does, and this name is used to “call” the function to perform its task when needed.
+
+## Defining and Calling Functions
+
+Any function has three important elements used to define it. First is the function name. This is used to call the function. Second is the number of parameters a function takes, parameters referring to the input the function takes when calling it. Lastly is the return type of the function, which refers to the output data type of the function. When defining the name of the function, the name should describe its functionality. 
+
+The figure below describes the syntax of functions in Swift:
+
+![This diagram explains the syntax of the functions in Swift ](https://raw.githubusercontent.com/Codecademy/docs/main/media/swift-functions-diagram.png)
+
+```Swift
+func salute(name: String) -> String {
+    let salute = "Hello, " + name + "!"
+    return salute
+}
+```
+
+For example, this function called `greet` takes a `String` parameter called `person`, and returns a `String` equal to "Hello" plus the `person` value. The `greet(person:)` function is called by passing it a `String` value after the `person:` argument label as shown below:
+
+```Swift
+print(salute(name: "Anna"))
+// Prints "Hello, Anna!"
+```
+
+## Multiple Parameters
+
+Functions can take more than one input parameter, the parameters must be separated by commas and assigned during the function call.
+For example, this function takes two numbers as an input, and returns the division of the two numbers:
+
+```Swift
+func sum(x: Int, y: Int) -> Int {
+  return x + y
+} 
+
+let result = sum(x: 1, y: 2) 
+print(result)
+ // Prints:  3
+```
+
+## Returning Multiple Values
+
+A tuple type can be used as the return type for a function to return multiple values as part of one compound return value. Each value within its parentheses needs to be labeled and assigned a type in the function definition.
+
+For example, this function takes array of `Int` as an input, and returns the smallest and largest number in the array.
+
+```Swift
```

Comment:
This example has not been updated and is a direct copy from the swift documentation.

Reply:
I changed , I wrote the code myself to avoid having the same problem, it just solves the same problem. @caupolicandiaz

Reply:
I updated the second block to use an original variable name and a different array